internal/pod_client_test.go

```diff
@@ -58,4 +62,47 @@ var _ = Describe("pod_client", func() {
 			Expect(podHasSidecarTLS(pod)).To(BeTrue())
 		})
 	})
+
+	When("generating a request", func() {
+		var retryClient *retryablehttp.Client
+		var target url.URL
+		getTimeout := 1 * time.Second
+		postTimeout := 10 * time.Second
+
+		BeforeEach(func() {
+			retryClient = retryablehttp.NewClient()
+			target = url.URL{
+				Scheme: "http",
+				Host:   "127.0.0.1:8080",
+				Path:   "test",
+			}
+		})
+
+		When("generating a http get request", func() {
+			It("should generate the request", func() {
+				req, err := generateRequest(retryClient, target.String(), http.MethodGet, getTimeout, postTimeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(req.Method).To(Equal(http.MethodGet))
+				Expect(retryClient.HTTPClient.Timeout).To(Equal(getTimeout))
+			})
+		})
+
+		When("generating a http post request", func() {
+			It("should generate the request", func() {
+				req, err := generateRequest(retryClient, target.String(), http.MethodPost, getTimeout, postTimeout)
```

Comment:
generateRequestSafe function that incorporates all the checks would trim down all the test code a bit?